src/message/commands/privmsg.rs

```diff
@@ -66,6 +70,15 @@ impl TryFrom<IRCMessage> for PrivmsgMessage {
 
         let (message_text, is_action) = source.try_get_message_text()?;
 
+        let (is_redeem, custom_reward_id) = {
+            let msg_id = source.try_get_optional_tag_value("msg-id")?.to_owned();
```

Comment:
Is this really correct? Do messages with `@msg-id` actually exist? (Mind you, not `@msg-id=` or `@msg-id=some-value`)?

Because if they don't exist, the parser method for that variant already exists (`try_get_optional_nonempty_tag_value`)
 
Apart from that, the method you created already returns `String`, not `&str`, so if we end up keeping it, I'd like to have `try_get_optional_tag_value` return `&str`, not `String`.

Reply:
I change return type to &str but i need to map Option<&str> to Option<String>. If you have other idea don't hesitate because i not perfectly know rust ^^. When i try change Option<String> to Option<&str> in pivmsg.rs struct. I must add lifetime in struct.

Reply:
I check the method `try_get_optional_nonempty_tag_value`. But it returns an error if tag is void and don't make that.
If msg-id exist but it's avoided, i think it's useless to generate parsing error just for that when we can continue without any problem.